### ISSUE TITLE
[M0-Dev] Estructuras necesarias

### ISSUE THREAD

**Issue by ignaciotitos** [RafaelT00/AutoIV#9]
Entiendo que se deberían de crear las clases Empleado y Vacaciones si entiendo bien en el problema. Si me falta algo o lo ves de otra manera coméntamelo.

**Comment by RafaelT00** [comment]
Al principio había pensado en dos estructuras, una para los empleados y otra para la información de la empresa(fechas de proyectos, restricciones entre empleados, etc), luego pensé que quizás es más facil manejar las restricciones entre empleados en una nueva estructura.

**Comment by ignaciotitos** [comment]
A que te refieres con las restricciones entre empleados?

**Comment by ignaciotitos** [comment]
Que clases crearías tu entonces? 

La entidad sería Empleado mientras que los objetos valor serían Vacaciones con las vacaciones disponibles y tal y Restricciones con las restricciones que pueda haber entre los empleados? O como

**Comment by RafaelT00** [comment]
> A que te refieres con las restricciones entre empleados?

Algunos empleados no se pueden marchar a la par porque es necesario que al menos este uno de los dos para que se puedan realizar ciertas tareas.

**Comment by ignaciotitos** [comment]
Entonces dentro de cada empleado tendría unas restricciones, o crearías una clase distinta con restricciones

**Comment by ignaciotitos** [comment]
Me lo imagino como un enlazador que conecte los distintos empleados, y que cuando uno está de vacaciones, otro no puede estarlo, y que sea un atributo común y conectado en la clase Empleado.

**Comment by RafaelT00** [comment]
Sí, algo así había pensado, creo que con eso es más fácil gestionar a los empleados.

**Comment by ignaciotitos** [comment]
Vale perfecto

**Comment by ignaciotitos** [comment]
Entonces va a tener la entidad Empleado y el objeto valor Vacaciones, no?

**Comment by RafaelT00** [comment]
Vacaciones no lo había pensado pero creo que sí podría ser util que cada empleado tenga un enlace a su objeto de Vacaciones. También quiero crear una estructura para almacenar lo que serían los proyectos y los empleados asociados a este. Inicialmente tenía pensado una estructura que funcionase como un calendario donde tendría los periodos de los trabajos y sobre el cual también añadiria los periodos de vacaciones, pero creo que sería poco manejable.

**Comment by ignaciotitos** [comment]
Es posible que ese struct sea Vacaciones, en el que se tienen los días de vacaciones de un empleado, los proyectos en los que está ese empleado, los proyectos a los que pertenece el empleado y con quien los comparte y el periodo de trabajo que comprenda. Te cuadra?

**Comment by ignaciotitos** [comment]
Cada Empleado tenga un Vacaciones asociado

**Comment by RafaelT00** [comment]
Sí, pero creo que los proyectos a los que pertenece el empleado los debería almacenar en Restricciones o en una estructura Proyectos.

**Comment by ignaciotitos** [comment]
Vale, entonces sería Empleado, Vacaciones y Proyectos/Restricciones, no?

**Comment by RafaelT00** [comment]
Diría que sí, aunque no estoy del todo segur de si debería separar Proyectos y Restricciones porque se podrían considerar restricciones que no tengan que ver con algun proyecto en concreto sino que sean por otro tipo de funciones que tengan los empleados dentro de la empresa.

**Comment by ignaciotitos** [comment]
Entonces por lo que entiendo Restricciones engloba a Proyectos, y Proyectos es como una herencia de Restricciones por así decirlo. Es decir, que en todo caso crearía solo Restricciones, ya que este engloba a todas las restricciones que se puedan tener.

**Comment by RafaelT00** [comment]
No lo había pensado  exáctamente así pero me gusta de esta manera porque con ello puedo almacenar para cada empleado lo justo necesario para decidir que días puede tener de vacaciones.

**Comment by ignaciotitos** [comment]
Pues sí la verdad. Entonces nos quedamos con Empleado, Vacaciones y Restricciones. ¿No?

**Comment by RafaelT00** [comment]
Sí, creo que es lo mejor.

**Comment by ignaciotitos** [comment]
Okeyy, perfect

**Comment by ignaciotitos** [comment]
Vuelvo a este issue. Veo que hay que enlazar la historia de usuario. Según la historia de usuario #7, para satisfacerla es necesario crear una clase Empleado, como ya habíamos hablado, y una clase AsignarVacaciones, que será parecida a la que tenemos ahora msimo. En cualquier caso Restricciones no creo que sea necesario ahora mismo. En todo caso dentro de AsignarVacaciones que se lleve la cuenta de peticiones, o en Empleado que se lleve la cuenta. Dime como lo ves.

**Comment by ignaciotitos** [comment]
También Empleado creo que sería un objeto valor y Vacaciones o AsignarVacaciones que sea la entidad.

**Comment by RafaelT00** [comment]
Creo que es mejor llevarla en Empleados y dejar AsignarVacaciones para hacer la asignación como tal.

**Comment by ignaciotitos** [comment]
Vale. Algo más que veas necesario para el milestone 0?

**Comment by ignaciotitos** [comment]
Veo que en el milestone 0 dices que necesitas una estructura para los proyectos y restricciones y otra para el calendario. Algo que realmente no se menciona en la HU. No entiendo realmente que necesitas exactamente para superar este objetivo.

**Comment by ignaciotitos** [comment]
En el milestone 0 necesitas un calendario donde aparecen las asignaciones.

**Comment by ignaciotitos** [comment]
closes #9 por mal planteamiento del issue

**Comment by ignaciotitos** [comment]
closes #9 por mal planteamiento del issue

**Comment by ignaciotitos** [comment]
Necesito que pongas este issue como inválido.
